plugins/modules/foreman_hostgroup.py

```diff
@@ -408,12 +419,36 @@ def main():
 
     parameters = entity_dict.get('parameters')
 
+    puppetclasses = entity_dict.pop('puppetclasses', None)
+    current_puppetclasses = entity.pop('puppetclasses', None) if entity else []
+
     changed, hostgroup = module.ensure_entity('hostgroups', entity_dict, entity)
 
     if hostgroup:
         scope = {'hostgroup_id': hostgroup['id']}
         changed |= module.ensure_scoped_parameters(scope, entity, parameters)
 
+    if not module.desired_absent and 'environment' in entity_dict:
```

Comment:
Why is this not `'environment_id' in hostgroup`?

Reply:
Because param name in this module is `environment` and there is `environment_id` in entity_dict, it appears only after parsing of params is done in foreman_helper

Reply:
But you are using it as `hostgroup['environment_id']` in line 435. So i thought, this would be the natural check.
Or can it appear in `hostgroup` even if it was not specified in `entity_dict`? And in that case we do _not_ want to act?

Reply:
We dont want to act is there is no environment, as then assigning classes is pointless.. But you are right, checking for hostgroup[env_id] feels safer in this case